tt_metal/hw/inc/debug/ring_buffer.h

```diff
@@ -9,7 +9,7 @@
 // 0 and immediately use it as an index for the first write.
 constexpr static int16_t DEBUG_RING_BUFFER_STARTING_INDEX = -1;
 
-#if defined(KERNEL_BUILD)
+#if defined(KERNEL_BUILD) || defined(FW_BUILD)
```

Comment:
do we need this?

Reply:
This was added to keep it safe if accidentally included from host-side code, adding the `defined(FW_BUILD)` is a bugfix. @pgkeller is the intent to include this clause on all device-side headers?

Reply:
We don't have a simple consistent mechanism for separating interface from implementation for host included headers.  This is a stop gap for now.  The fix is that both FW and KERNEL are ok, not just KERNEL

Reply:
Yeah just wondering if I should add this clause to all watcher/dprint headers?